### ISSUE TITLE
Generating client for Enode API fails

### ISSUE THREAD

**Issue by bartoszirzyk** [apple/swift-openapi-generator#221]
Xcode 15.0 beta 6
swift-open-api-generator: 0.1.11

Hi,
using versions mentioned above I tried to generate a client for [EnodeAPI](https://enode-api.production.enode.io/openapi/latest.json). It fails with reason:

```
Schema "oneOf ()" is not supported, reason: "not an object-ish schema (object, ref, allOf)", skipping [context: foundIn=#/components/schemas/TemperatureTargetState]
warning: Schema "oneOf ()" is not supported, reason: "not an object-ish schema (object, ref, allOf)", skipping [context: foundIn=Components.Schemas.PartialTemperatureSchedule (#/components/schemas/PartialTemperatureSchedule)/defaultTargetState]
warning: Schema "oneOf ()" is not supported, reason: "not an object-ish schema (object, ref, allOf)", skipping [context: foundIn=Components.Schemas.PartialTemperatureSchedule.rulesPayloadPayload.Value2Payload (#/components/schemas/PartialTemperatureSchedule/rulesPayload/value2)/targetState]
Error: In discriminator.mapping, found multiple keys for the same value.
```

**Comment by czechboy0** [comment]
Hi @bartoszirzyk, thanks for taking the time to file this issue! 

The formatting isn't great here, but there are actually 4 issues here.

The first 3 are warnings:
```
Schema "oneOf ()" is not supported, reason: "not an object-ish schema (object, ref, allOf)", skipping [context: foundIn=#/components/schemas/TemperatureTargetState]
warning: Schema "oneOf ()" is not supported, reason: "not an object-ish schema (object, ref, allOf)", skipping [context: foundIn=Components.Schemas.PartialTemperatureSchedule (#/components/schemas/PartialTemperatureSchedule)/defaultTargetState]
warning: Schema "oneOf ()" is not supported, reason: "not an object-ish schema (object, ref, allOf)", skipping [context: foundIn=Components.Schemas.PartialTemperatureSchedule.rulesPayloadPayload.Value2Payload (#/components/schemas/PartialTemperatureSchedule/rulesPayload/value2)/targetState]
```

This is a genuine bug in the generator, as we don't currently support nesting oneOfs. I'll file a separate issue to track that, but it'll only make sense to fix once #144 is landed, so marking this blocked for now.

Separately, there seems to be a fatal error, which is why the generator fails:

```
Error: In discriminator.mapping, found multiple keys for the same value.
```

This seems to be caused by another generator bug, this time because the discriminator mapping's values aren't unique. I don't remember why it's enforced, and it seems incorrect, so I'll file a separate issue as well.

I'll keep this issue open until the two blocking issues get resolved, to make it easier to track once this OpenAPI doc can be generated.

**Comment by czechboy0** [comment]
Okay, @bartoszirzyk - I just confirmed that with Swift OpenAPI Generator 0.1.13, the Enode OpenAPI document produces compiling code.

Note that you'll need to enable all these feature flags, e.g. in your `openapi-generator-config.yaml`, by adding:

```yaml
featureFlags:
  - proposal0001
  - multipleContentTypes
  - strictOpenAPIValidation
  - closedEnumsAndOneOfs
```

They will become the default in the upcoming 0.2.0 as well, at which point you won't need to explicitly specify them.

But you should be unblocked now, please file new issues if you find anything else.

**Comment by bartoszirzyk** [comment]
Perfect, thanks a lot @czechboy0
